### ISSUE TITLE
Enable iGPU?

### ISSUE THREAD

**Issue by SolarJuice** [mikeeq/mbp-fedora#17]
Hello,
I tried using rEFInd / 0xbb's apple_set_os.efi to spoof the os version, and tried 0xbb's gpu-switch to swtich graphics. But it seems that his gpu-switch script doesn't work for this distro / mbp15,1.
https://github.com/0xbb/gpu-switch/issues/51
I've put an issue under his script's page, but I think it is also a good idea to ask it here as it may be a distro compatibility issue - "Couldn't mount 'efivarfs' " (see the above issue)

Above all, any workaround to enable iGPU screen rendering? because I need to run KVM virtual machine and passthrough the dGPU to a VM. Many thanks!

**Comment by mikeeq** [comment]
Hi @SolarJuice,
After macOS Catalina update it's no longer possible to mount efivarfs - mac just freezes whenever you're trying to do it.

**Comment by SolarJuice** [comment]
> Hi @SolarJuice,
> After macOS Catalina update it's no longer possible to mount efivarfs - mac just freezes whenever you're trying to do it.

Thanks for the answer. Is this a firmware update that cannot roll back? Does downgrading to Mojave work?

**Comment by mikeeq** [comment]
I don't think that the downgrade process will downgrade the version of EFI itself, but I haven't tested it - so if you want, you can try it ;)

**Comment by cristianmiranda** [comment]
Hi @SolarJuice @mikeeq I'm having exactly the same issue. Is there a way to workaround this?.

**Comment by encombhat** [comment]
See this post for a solution: https://apple.stackexchange.com/questions/166876/macbook-pro-how-to-disable-discrete-gpu-permanently-from-efi

TL;dr: Execute this command to set efivars manually in macOS single user mode: 
`nvram fa4ce28d-b62f-4c99-9cc3-6815686e30f9:gpu-power-prefs=%01%00%00%00`

Works for me (Macbook Pro 15', 2019)

**Comment by SolarJuice** [comment]
@encombhat Thanks! I'll try it out. btw is this permanant or revertable?

**Comment by encombhat** [comment]
> @encombhat Thanks! I'll try it out. btw is this permanant or revertable?

Well, it should be persistent across reboots, but you can always reset the NVRAM manually.

**Comment by icyLollipop** [comment]
> > @encombhat Thanks! I'll try it out. btw is this permanant or revertable?
> 
> Well, it should be persistent across reboots, but you can always reset the NVRAM manually.

cool, thanks~
